File: app/src/main/java/com/damhoe/scoresheetskat/score/application/ports/in/CreateScoreUseCase.java
package com.damhoe.scoresheetskat.score.application.ports.in;

import com.damhoe.scoresheetskat.base.Result;
import com.damhoe.scoresheetskat.score.domain.SkatScore;
import com.damhoe.scoresheetskat.score.domain.SkatScoreCommand;

public interface CreateScoreUseCase {
 Result<SkatScore> createScore(SkatScoreCommand command);
 Result<SkatScore> updateScore(long id, SkatScoreCommand command);
 Result<SkatScore> deleteScore(long id);
}

File: app/src/main/java/com/damhoe/scoresheetskat/score/application/ports/in/CreateScoreUseCase.kt
package com.damhoe.scoresheetskat.score.application.ports.`in`

import com.damhoe.scoresheetskat.score.domain.SkatScore
import com.damhoe.scoresheetskat.score.domain.SkatScoreCommand

interface CreateScoreUseCase {
    fun createScore(command: SkatScoreCommand): Result<SkatScore>
    fun updateScore(id: Long, command: SkatScoreCommand): Result<Unit>
    fun deleteScore(id: Long): Result<SkatScore>
    fun deleteScoresForGame(gameId: Long): Result<Int>
}